redis/src/main/scala/zio/redis/PubSubCommand.scala

```diff
@@ -0,0 +1,29 @@
+package zio.redis
+
+import zio.schema.codec.BinaryCodec
+import zio.stream._
+import zio.{IO, ZLayer}
+
+final case class RedisPubSubCommand(command: PubSubCommand, codec: BinaryCodec, executor: RedisPubSub) {
+  def run: IO[RedisError, List[Stream[RedisError, PushProtocol]]] = {
+    val codecLayer = ZLayer.succeed(codec)
+    executor.execute(command).provideLayer(codecLayer)
+  }
+}
+
+sealed trait PubSubCommand
+
+object PubSubCommand {
+  case class Subscribe(
+    channel: String,
+    channels: List[String],
+    onSubscribe: PubSubCallback
+  ) extends PubSubCommand
+  case class PSubscribe(
+    pattern: String,
+    patterns: List[String],
+    onSubscribe: PubSubCallback
+  ) extends PubSubCommand
+  case class Unsubscribe(channels: List[String])  extends PubSubCommand
+  case class PUnsubscribe(patterns: List[String]) extends PubSubCommand
+}
```

Comment:
We should hide these from the end user. These should be in a dedicated `PubSubCommand` file.

Reply:
I moved it to its own file but do we need to add package private access modifier?

Reply:
I think that we do if it's not part of the `api` interface.

Reply:
I added package private accessor